### PR TITLE
Enable dependabot to keep submodules up-to-date

### PR DESCRIPTION
**Pull request by nwiltsie**
# Description
This enables [dependabot](https://github.com/dependabot) for this repository, specifically to keep git submodules up-to-date. Once merged, dependabot will automatically open pull requests to update any submodules to the latest versions. Thereafter, it will check weekly to see if new commits have been made to those submodules, opening new PRs as necessary.

See https://github.com/uclahs-cds/group-nextflow-working-group/issues/50 for more context for these changes.

No PRs, including this one and all dependabot PRs, will have the power to make updates without human intervention. The changes will still need to be tested manually before they should be merged.

### CODE REVIEW

**Review by sorelfitzgibbon — APPROVED** [review]

**Review by yashpatel6 — APPROVED** [review]

**Comment by zhuchcn** [comment]
I'm just wondering what will be the best practice to use the PRs opened by dependabot. Directly merging them without testing doesn't sound very secure to me. So once a PR is opened by dependabot for submodule update, should someone test it before merging the PR?

**Comment by nwiltsie** [comment]
Yes, absolutely someone should test it the changes. I did it for pipeline-recalibrate-BAM like so:

https://github.com/uclahs-cds/pipeline-recalibrate-BAM/pull/41#pullrequestreview-1773030938
https://github.com/uclahs-cds/pipeline-recalibrate-BAM/pull/42#pullrequestreview-1773104326

**Comment by zhuchcn** [comment]
Ah yeah you also said this in the description of this PR. I should have finished reading the whole message.

**Comment by sorelfitzgibbon** [comment]
I have a couple questions.   
1. Should I merge this PR now?
2. How do I test the dependabot branches?    
For `pipeline-SQC-DNA`, from within the branch I'm developing, I tried:
 - `git pull origin dependabot/submodules/external/pipeline-Nextflow-module-b0b2798`
 - `git pull origin dependabot/submodules/external/pipeline-Nextflow-config-1960eaa`
but the submodules still seem to be on an older commit.  
Guidance appreciated!

**Comment by nwiltsie** [comment]
> I have a couple questions.   
> 1. Should I merge this PR now?

Yes, there's no harm in doing so - nothing will change other than dependabot will start opening other PRs (which you can ignore if you want to).

> 2. How do I test the dependabot branches?    
> For `pipeline-SQC-DNA`, from within the branch I'm developing, I tried:
>  - `git pull origin dependabot/submodules/external/pipeline-Nextflow-module-b0b2798`
>  - `git pull origin dependabot/submodules/external/pipeline-Nextflow-config-1960eaa`
> but the submodules still seem to be on an older commit. 

Using https://github.com/uclahs-cds/pipeline-recalibrate-BAM/pull/44 as an example, here's my workflow from within an existing checkout:
1. `git fetch origin`
2. `git checkout dependabot/submodules/external/pipeline-Nextflow-config-1960eaa` (I just copy the branch name from the pull request - there's a handy little button at the top to do so automatically)
3. `git submodule update --init --recursive` **This** is the worst thing about submodules - you have you update them manually
4. Test as appropriate!
5. After going back to whatever branch you were on before, run `git submodule update --init --recursive` _again_ to restore what you had previously.

Basically submodules act lazily - unlike all other files, when you `checkout` a new commit they will _not_ update automatically.

**Comment by sorelfitzgibbon** [comment]
Thanks @nwiltsie!
